subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/TestDataGenerator.java

```diff
@@ -163,4 +173,43 @@ public String getDate() {
             return commits;
         }
     }
+
+    static class BackgroundColor {
+        // https://www.w3schools.com/colors/colors_picker.asp
+        private static final String RED = "#ff0000";
+        private static final String ORANGE = "#ff8000";
+        private static final String LIGHT_GREEN = "#80ff00";
+        private static final String GREEN = "#00ff00";
+        private Map xaxis;
+        private String color;
+
+        private static BackgroundColor ofConfidence(List<Number> xy) {
+            double index = xy.get(0).doubleValue();
+            double confidencePercentage = xy.get(1).doubleValue();
+            if (confidencePercentage > 90) {
+                return new BackgroundColor(index, RED);
```

Comment:
I would have preferred to use a linear background color, so something like
```
int limit = 80
int aboveLimit = confidencePercentage - limit;
int redValue = aboveLimit * 256 / (100 - limit);
```
And then use the `redValue` for the red component of the background. Same goes for the green value for improvements.

We can also start out with what you have right now and improve later on.